jerry-core/jerry.c

```diff
@@ -591,6 +591,47 @@ jerry_get_array_index_value (jerry_object_t *array_obj_p, /**< array object */
 } /* jerry_get_array_index_value */
 
 /**
+ * Get length of an array object
+ *
+ * @return length of the given array
+ */
+uint32_t
+jerry_get_array_length (const jerry_object_t *object_p)
+{
+  jerry_assert_api_available ();
+
+  JERRY_ASSERT (jerry_is_array (object_p));
```

Comment:
Basically what I was thinking is, since any object can behave as an array as long as it has a length field, people may end giving specialized objects to this function thinking that it will work every time (but may fail in some extreme circumstances). On the other hand, seeing that other functions also behave in the way, that giving incorrect arguments will yield undefined behavior, it's probably fine to have this do the same. (But I would definitely leave note about it in the documentation)